File: ReverseInteger.py
"""
7. Reverse Integer
Reverse digits of an integer.
Example1: x = 123, return 321
Example2: x = -123, return -321
"""

class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        flag = 1
        if x < 0:
            flag = -1
            s = str(x)[1:]
        else:
            s = str(x)
        i, res = 0, 0
        while i < len(s):
            res += int(s[i]) * 10 ** i
            i += 1
        res *= flag
#Actually, for python, the res won't be overflow
        if res < -2 ** 31 or res > 2 ** 31 - 1:
            return 0
        return res

if __name__ == "__main__":
    x1 = -123
    x2 = 2**31 - 1
    print(Solution().reverse(x1))
    print(Solution().reverse(x2))

File: Java/ReverseInteger.java

public class ReverseInteger {
    public int reverse(int x) {
        if(x < 0) {
            if(x == Integer.MIN_VALUE) {return 0;}
            return -reverse(-x);
        }
        double res = 0;
        while(x != 0) {
            res = res * 10 + x % 10;
            x /= 10;
        }
        if(res < Integer.MIN_VALUE || res > Integer.MAX_VALUE) {return 0;}
        return (int)res;
    }

}
